### ISSUE TITLE
Failure to Build: No rule to make target '../genie/libgenie.a'

### ISSUE THREAD

**Issue by Enverex** [FolkertVanVerseveld/aoe#11]
Doesn't seem to build anymore (from a clean sync and build)...

`make[1]: *** No rule to make target '../genie/libgenie.a', needed by 'cheat'.  Stop.`

I assume it's related to https://github.com/FolkertVanVerseveld/aoe/commit/bff4917aca90112d7852115397991fab2691a4ac

**Comment by FolkertVanVerseveld** [comment]
Hello Enverex,

Thank you for testing the build scripts. I assume your local repo points 
to commit 9d8614c. Are you sure you ran make in src/?

The building process has a bit changed indeed. The tools, the genie game
engine and the game itself are separated which means the tools and the
game depend on the .a file created in src/genie. I will update the
building documentation to make this more clear.

If you can't get it to build, you could try the develop branch as I've 
pushed some more changes there (it should not make any difference, but
maybe it does).

Your feedback is greatly appreciated, so please don't hesitate to give
more feedback if instructions are unclear or if something is missing.

All the best,

Folkert van Verseveld

**Comment by FolkertVanVerseveld** [comment]
Hmmm, that's weird. I always run make in parallel with other projects I
try to compile, but not my own ones... I totally forgot about that might 
being an issue. I will investigate this and see if the makefile is 
malformed for running in parallel.

**Comment by FolkertVanVerseveld** [comment]
I've reproduced your issue when running make in parallel. It is still
possible to make the project in parallel, but you have to run it in
using these steps:

First, run ./configure in src/ to setup all directories, then run make 
in src/genie and finally run make -jXX in src/ to make the other files.

**Comment by FolkertVanVerseveld** [comment]
Currently, I've partially fixed running make in parallel. The current
workaround is to run ./configure and then ./make_parallel where you can
optionally provide the number of cores (default 8). E.g.:

  ./configure && ./make_parallel 4
  
Configures the project and makes it using 4 cores. Note that all source
has been moved from src/ to the project root folder. This workaround is
merged into master.

I'm going to either eliminate the need to let make run make recursively
to fix these parallel make errors or use a different build system that
can resolve these issues. I'm closing this issue for now, but feel free
to reopen it in case it does not work.
